packet_test.go

```diff
@@ -1129,3 +1129,256 @@ func BenchmarkMarshalTo(b *testing.B) {
+					}},
+					{3, []byte{
+						0xCC, 0xCC, 0xCC, 0xCC,
+					}},
+				},
+				Version:        2,
+				PayloadType:    96,
+				PayloadOffset:  28,
+				SequenceNumber: 27023,
+				Timestamp:      3653407706,
+				SSRC:           476325762,
+				CSRC:           []uint32{},
+			},
+				Payload: []byte{
+					0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+				Raw: []byte{
+					0x90, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+					0x27, 0x82, 0x10, 0x00, 0x00, 0x03, 0x01, 0x00, 0x02, 0x01,
+					0xBB, 0x00, 0x03, 0x04, 0xCC, 0xCC, 0xCC, 0xCC, 0x98, 0x36,
+					0xbe, 0x88, 0x9e,
+				},
+			},
+		},
+		"DefaultExtension": {
+			rawPkt: []byte{
+				0x80, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+				0x27, 0x82, 0x00, 0x01, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+			},
+			expectedPacket: &Packet{Header: Header{
+				Marker:           true,
+				Extension:        false,
+				ExtensionProfile: 0,
+				Version:          2,
+				PayloadOffset:    12,
+				PayloadType:      96,
+				SequenceNumber:   27023,
+				Timestamp:        3653407706,
+				SSRC:             476325762,
+				CSRC:             []uint32{},
+			},
+				Payload: []byte{
+					0x00, 0x01, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+				Raw: []byte{
+					0x80, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+					0x27, 0x82, 0x00, 0x01, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+			},
+		},
+	}
+
+	for name, testCase := range cases {
+		t.Run(name, func(t *testing.T) {
+			p := &Packet{}
+			// First Unmarshal
+			if err := p.Unmarshal(testCase.rawPkt); err != nil {
+				t.Error("Test multiple unmarshal: first: %w", err)
+			} else if len(p.Extensions) != len(testCase.expectedPacket.Extensions) {
+				t.Errorf("Test multiple unmarshal: first: got %d extensions, want %d", len(p.Extensions), len(testCase.expectedPacket.Extensions))
+			} else if !reflect.DeepEqual(p, testCase.expectedPacket) {
+				t.Errorf("TestMultiple unmarshal: first: got %#v, want %#v", p, testCase.expectedPacket)
+			}
+
+			// Second Unmarshal
+			if err := p.Unmarshal(testCase.rawPkt); err != nil {
+				t.Error("Test multiple unmarshal: second: %w", err)
+			} else if len(p.Extensions) != len(testCase.expectedPacket.Extensions) {
+				t.Errorf("Test multiple unmarshal: second: got %d extensions, want %d", len(p.Extensions), len(testCase.expectedPacket.Extensions))
+			} else if !reflect.DeepEqual(p, testCase.expectedPacket) {
+				t.Errorf("TestMultiple unmarshal: second: got %#v, want %#v", p, testCase.expectedPacket)
+			}
+		})
+
+	}
+
+}
+
+func TestUnmarshalExtensionThenPacketWithoutExtension(t *testing.T) {
+	pktWithExtensions := []byte{
+		0x90, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+		0x27, 0x82, 0xBE, 0xDE, 0x00, 0x01, 0x50, 0xAA, 0x00, 0x00,
+		0x98, 0x36, 0xbe, 0x88, 0x9e,
+	}
+
+	pktWithoutExtensions := []byte{
+		0x80, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+		0x27, 0x82, 0x00, 0x01, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+	}
+
+	parsedPacket := &Packet{Header: Header{
```

Comment:
Suggestion - call the variable packetWithoutExtensions to make it clear.
Also add a comment like:
 
  // Directly create an instance of a Packet that is equivalent to the pktWithoutExtensions.  Most 
  // notably this Packet does not have an extension which distinguishes it from pktWithExtensions.